atc/exec/task_config_source.go

```diff
@@ -79,7 +80,10 @@ func (configSource FileConfigSource) FetchConfig(logger lager.Logger, repo *arti
 		return atc.TaskConfig{}, UnknownArtifactSourceError{sourceName, configSource.ConfigPath}
 	}
 
-	stream, err := source.StreamFile(logger, filePath)
+	// TODO: idk if we really want to pass a context in here. This is a very
+	// 		 shortlived stream. It would pollute the TaskConfigSource Interface
+	//		 as all the FetchConfigs will be need to have this passed in.
+	stream, err := source.StreamFile(context.TODO(), logger, filePath)
```

Comment:
look at this note and tell me if you agree

Reply:
I'm cool with leaving it out for now, as long as we have a comment explaining why

Reply:
I'm leaning towards just implementing it. TaskConfigSource is only used by Tasks, right? You could also use `lagerctx` to pass the `lager.Logger` argument through the `Context` argument if you don't want to add another argument.

It's a short-lived stream but if we're intending to guard against an unreachable worker, it won't really matter how long the transfer takes. This is also one of the first things that happens in a task step, so it could be the first thing that hangs.

Reply:
done